barman/recovery_executor.py

```diff
@@ -1807,13 +1810,317 @@ def _backup_copy(self, backup_info, dest, remote_command=None, **kwargs):
+            the base backup, in case of remote backup.
+        :return dict: ``recovery_info`` dictionary, holding the values related
+            with the recovery process.
+        """
+        # First combine the backups, generating a new synthetic backup in the staging area
+        combine_directory = self.config.recovery_staging_path
+        synthetic_backup_info = self._combine_backups(backup_info, combine_directory)
+
+        # Add the backup directory created in the staging area to be deleted after recovery
+        synthetic_backup_dir = synthetic_backup_info.get_basebackup_directory()
+        self.temp_dirs.append(fs.LocalLibPathDeletionCommand(synthetic_backup_dir))
+
+        # Perform the standard recovery process passing the synthetic backup
+        return super(IncrementalRecoveryExecutor, self).recover(
+            synthetic_backup_info, dest, remote_command=remote_command, **kwargs
+        )
+
+    def _combine_backups(self, backup_info, dest):
+        """
+        Combines the backup chain into a single synthetic backup using the
+        ``pg_combinebackup`` utility.
+
+        :param barman.infofile.LocalBackupInfo backup_info: the incremental
+            backup to be recovered
+        :param str dest: the directory where the synthetic backup is going
+            to be mounted on
+        :return barman.infofile.SyntheticBackupInfo: the backup info file of the
+            combined backup
+        """
+        self.combine_start_time = datetime.datetime.now()
+
+        # Build the synthetic backup info from the incremental backup as it has
+        # the most recent data relevant to the recovery. Also, the combine process
+        # should be transparent to the end user so e.g. the .barman-recover.info file
+        # that is created on destination and also the backup_id that is appended to the
+        # manifest file in further steps of the recovery should be the same as the incremental
+        synthetic_backup_info = SyntheticBackupInfo(
+            self.server,
+            base_directory=dest,
+            backup_id=backup_info.backup_id,
+        )
+        synthetic_backup_info.load(filename=backup_info.filename)
+
+        dest_dirs = [synthetic_backup_info.get_data_directory()]
+
+        # Maps the tablespaces from the old backup directory to the new synthetic
+        # backup directory. This mapping is passed to the pg_combinebackup as input
+        tbs_map = {}
+        if backup_info.tablespaces:
+            for tablespace in backup_info.tablespaces:
+                source = backup_info.get_data_directory(tablespace_oid=tablespace.oid)
+                destination = synthetic_backup_info.get_data_directory(
+                    tablespace_oid=tablespace.oid
+                )
+                tbs_map[source] = destination
+                dest_dirs.append(destination)
+
+        # Prepare the destination directories for pgdata and tablespaces
+        for d in dest_dirs:
```

Comment:
let's avoid using single char vars... I know that this happens in the already existing code, but is wrong and I would like to avoid this.
```suggestion
        for dir in dest_dirs:
```